### ISSUE TITLE
CS8602 false positive

### ISSUE THREAD

**Issue by Logerfo** [dotnet/roslyn#40724]
## Version Used
Visual Studio 2019 (16.5.0 Preview 1.0)

## Steps to Reproduce
```c#
#nullable enable
public class C {
    bool M() {
        object? newValue = null;
        object? oldValue = null;
        return newValue is null ^ oldValue is null || !(newValue is null && oldValue is null || newValue.Equals(oldValue));
    }
}
```
[SharpLab](https://sharplab.io/#v2:CYLg1APgxAdgrgGwQQwEYIKYAIMzZgWACgABAZixICYsBhLAb2Kxa1QHt2EsBZACgCUjZq1HtUAKwwBjAC4B+LDAwB3AGrIEcbAF4liBAG4RoluKlzFXYBq279SY0VOsSAdiWrb2rAEsAzg7cAHpY1t7YAUFYEBBYAIR8yuqaPlHwSFgAZFlhCDapkYEZ3LGeKXYAdACiAI5wmv584YUCAk6iAL7EnUA).

## Expected Behavior
Since `newValue` will never be `null` at the time of `.Equals` evaluation, no warning should be presented.

## Actual Behavior
> Warning CS8602: Dereference of a possibly null reference.

**Comment by jcouv** [comment]
I believe this is a known limitation in the nullable analysis of `^`. In the branch when `newValue is null ^ oldValue is null`, we know nothing about the nullability of `newValue` or `oldValue`; they may both be nullable in that branch.
